src/modules/dirwatcher.js

```diff
@@ -0,0 +1,140 @@
+import fs from 'fs';
+import AppEmitter from './../AppEmitter.js';
+
+export default class DirWatcher {
+  constructor() {
+    this.eventEmitter = new AppEmitter();
+  }
+
+  getEventEmitter() {
+    return this.eventEmitter;
+  }
+
+  watch (path, delay) {
+    let previousStats = [];
+
+    setInterval(() => {
+      fs.readdir(path, (err, newFiles) => {
+        let newStats = this.getStats(path, newFiles);
+
+        newStats.then(currentStats => {
+          if (err) {

```

Comment:
Maybe need to move error handling higher?

Reply:
Sure, thanks)